db/records.py

```diff
@@ -0,0 +1,55 @@
+from sqlalchemy import delete, select
+from sqlalchemy.inspection import inspect
+
+
+def _get_primary_key_column(table):
+    primary_key_list = list(inspect(table).primary_key)
+    # We do not support getting by composite primary keys
+    assert len(primary_key_list) == 1
+    return primary_key_list[0]
+
+
+def get_record(table, engine, id_value):
+    primary_key_column = _get_primary_key_column(table)
+    query = select(table).where(primary_key_column == id_value)
+    with engine.begin() as conn:
+        result = conn.execute(query).fetchall()
+        assert len(result) <= 1
+        return result[0] if result else None
+
+
+def get_records(table, engine, limit=None, offset=None):
+    query = select(table).limit(limit).offset(offset)
+    with engine.begin() as conn:
+        return conn.execute(query).fetchall()
+
+
+def create_records(table, engine, record_data):
```

Comment:
This function (through its call to `get_record` needs two connections (and two transactions) to insert a single record.  Getting those connections from the pool will bog this down under heavy load.

Why not just build the returned `Row` object (or named tuple) using the info in hand?  We'll know the insert was successful once we have the primary key in hand, and that's the only piece that we don't know beforehand.  Or, we might try to figure out how to batch multiple inserts for performance.

Reply:
This does batch multiple inserts, `get_record` is only called when there's a single insert. I put it under `engine.begin()` so there should only be one transaction/connection now.